api/src/main/java/org/apache/iceberg/io/LocationProvider.java

```diff
@@ -28,6 +28,16 @@
  * <p>Implementations must be {@link Serializable} because instances will be serialized to tasks.
  */
 public interface LocationProvider extends Serializable {
+
+  /**
+   * Return a fully-qualified data location for a table.
```

Comment:
Redundant javadoc, for these we usually just do a one liner, no annotation
```java
/**
Returns a fully-qualified data location for a table
**/
```

But I would also argue we should be a bit more descriptive, what is a "data location"

Additionally, do we want to have a default here? Shouldn't it be illegal to not provide this method? I think the intent here is not to break old location providers, but since the new code references this method I'm not sure we can get around it.

Reply:
Yes @RussellSpitzer I intent to not break custom location providers. There is noway I can provide default implementation also, thats why I am throwing exception.  Could you please recommend a right approach to follow here.

I added descriptive javadoc for both LocationProvider and MetadataLocationProvider. Thanks